File: Languages/Java/Day8.java
// Languages - Java - Day 8
// Foundational Concepts: Basics, syntax, loops, conditionals, functions, and variables.

// Create a program that takes a "user's" first and last name as input, concatenates them into a full name, slices the full name to extract specific characters, and formats the result in a greeting message.

import java.util.Scanner;

public class Day8 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.println("Enter your first name: ");
        String firstName = scanner.nextLine().trim();

        if (firstName.isEmpty()) {
            System.out.println("First name cannot be empty.");
            return;
        }

        System.out.println("Enter your last name: ");
        String lastName = scanner.nextLine().trim();

        if (lastName.isEmpty()) {
            System.out.println("Last name cannot be empty.");
            return;
        }

        String fullName = firstName + " " + lastName;

        char firstInitial = Character.toUpperCase(firstName.charAt(0));
        char lastInitial = Character.toUpperCase(lastName.charAt(0));

        String greeting = "Hello, " + fullName + "! Your initials are " + firstInitial + "." + lastInitial + ".";
        System.out.println(greeting);
    }
}

File: Languages/Rust/Day8.rs
// Languages - Rust - Day 8
// Foundational Concepts: Basics, syntax, loops, conditionals, functions, and variables.

// Create a program that takes a "user's" first and last name as input, concatenates them into a full name, slices the full name to extract specific characters, and formats the result in a greeting message.

use std::io;

fn main() {
    println!("Enter your first name:");
    let mut first_name = String::new();
    io::stdin().read_line(&mut first_name).expect("Failed to read input");
    let first_name = first_name.trim();

    if first_name.is_empty() {
        println!("First name cannot be empty.");
        return;
    }

    println!("Enter your last name:");
    let mut last_name = String::new();
    io::stdin().read_line(&mut last_name).expect("Failed to read input");
    let last_name = last_name.trim();

    if last_name.is_empty() {
        println!("Last name cannot be empty.");
        return;
    }

    let full_name = format!("{} {}", first_name, last_name);

    let first_initial = first_name.chars().next().unwrap_or(' ');
    let last_initial = last_name.chars().next().unwrap_or(' ');

    println!(
        "Hello, {}! Your initials are {}.{}.",
        full_name, first_initial, last_initial
    );
}
